domain/src/test/java/org/oppia/domain/classify/rules/ratioInput/RatioInputIsEquivalentRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,154 @@
+package org.oppia.domain.classify.rules.ratioInput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.RatioExpression
+import org.oppia.domain.classify.RuleClassifier
+import org.oppia.domain.classify.rules.ratioExpressionInput.RatioInputIsEquivalentRuleClassifierProvider
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [RatioInputIsEquivalentRuleClassifierProviderTest]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class RatioInputIsEquivalentRuleClassifierProviderTest {
+  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 0)
+  private val ITEM_RATIO_1 = listOf(1, 2, 3)
+  private val ITEM_RATIO_2 = listOf(2, 4, 6)
+
+  @Inject
+  internal lateinit var ratioInputIsEquivalentRuleClassifierProvider:
+    RatioInputIsEquivalentRuleClassifierProvider
+
+  private val isEquivalentClassifierProvider: RuleClassifier by lazy {
+    ratioInputIsEquivalentRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Test
+  fun testAnswer_testRatio_ratio2_bothValuesDoNotMatch() {
+    val inputs = mapOf("x" to createRatio(ITEM_RATIO_2))
+
+    val matches =
+      isEquivalentClassifierProvider.matches(answer = createRatio(ITEM_RATIO_2), inputs = inputs)
+
+    Truth.assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer_testRatio_ratio1_bothValuesMatch() {
+    val inputs = mapOf("x" to createRatio(ITEM_RATIO_1))
+
+    val matches =
+      isEquivalentClassifierProvider.matches(answer = createRatio(ITEM_RATIO_1), inputs = inputs)
+
+    Truth.assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer_testRatio_ratio2_bothValuesMatch() {
+    val inputs = mapOf("x" to createRatio(ITEM_RATIO_2))
+
+    val matches =
+      isEquivalentClassifierProvider.matches(answer = createRatio(ITEM_RATIO_1), inputs = inputs)
+
+    Truth.assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer_nonNegativeInput_inputWithIncorrectType_throwsException() {
+    val inputs = mapOf("x" to NON_NEGATIVE_VALUE_0)
+
+    val exception = assertThrows(IllegalStateException::class) {
+      isEquivalentClassifierProvider.matches(
+        answer = createRatio(ITEM_RATIO_1),
+        inputs = inputs
+      )
+    }
+
+    Truth.assertThat(exception)
```

Comment:
Directly import ``assertThat`` for consistency with other tests in the codebase.

Reply:
done